graphql/src/schema/ast.rs

```diff
@@ -104,6 +106,21 @@ pub fn get_field_type<'a>(object_type: &'a ObjectType, name: &Name) -> Option<&'
     object_type.fields.iter().find(|field| &field.name == name)
 }
 
+
+/// Returns the ValueType of a schema field
+pub fn get_value_type(schema_type: Type) -> Result<ValueType, ValueTypeError> {
+    match schema_type {
+        Type::NamedType(ref name) => Ok(ValueType::from_str(&name).unwrap()),
+        Type::NonNullType(inner) => match *inner {
+            Type::NamedType(ref name) => Ok(ValueType::from_str(&name).unwrap()),
+            Type::NonNullType(_) => Err(ValueTypeError::NestedNonNullType),
+            Type::ListType(_) => Err(ValueTypeError::CannotConvertFromListType),
```

Comment:
Nested non-null types [are not valid graphql](https://facebook.github.io/graphql/draft/#sec-Type-References) and I've tested that it's a parse error, so you can remove that error and replace this whole match with a recursive call to `get_value_type`.